tests/integration/test_charm.py

```diff
@@ -193,12 +195,13 @@ async def test_postgresql_parameters_change(ops_test: OpsTest) -> None:
                 )
                 records = cursor.fetchall()
                 settings = convert_records_to_dict(records)
-        connection.close()
 
-        # Validate each configuration set by Patroni on PostgreSQL.
-        assert settings["max_prepared_transactions"] == "100"
-        assert settings["shared_buffers"] == "128"
-        assert settings["lc_monetary"] == "en_GB.utf8"
+                # Validate each configuration set by Patroni on PostgreSQL.
+                assert settings["max_prepared_transactions"] == "100"
+                assert settings["shared_buffers"] == "128"
+                assert settings["lc_monetary"] == "en_GB.utf8"
```

Comment:
Check each node's config values.

Reply:
Thanks for catching that and improving it!